File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02781/p02781_num4_parsed.cpp
#include <iostream>
#include <string>
using namespace std;

int countNonZeroDigits(long long n) {
    int count = 0;
    string num = to_string(n);
    for (char digit : num) {
        if (digit != '0') {
            count++;
        }
    }
    return count;
}

int main() {
    long long N;
    int K;
    cin >> N >> K;
    
    int count = 0;
    for (long long i = 1; i <= N; i++) {
        if (countNonZeroDigits(i) == K) {
            count++;
        }
    }
    
    cout << count << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02781/p02781_num4_parsed.py
# Taking input from standard input
N = int(input())
K = int(input())

# Initializing the count
count = 0

# Looping through the numbers from 1 to N
for num in range(1, N+1):
    # Converting the number to string and counting the non-zero digits
    num_str = str(num)
    non_zero_digits = sum(1 for digit in num_str if digit != '0')

    # Incrementing the count if the number has exactly K non-zero digits
    if non_zero_digits == K:
        count += 1

# Printing the count
print(count)